aws/resource_aws_guardduty_publishing_destination.go

```diff
@@ -0,0 +1,182 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/guardduty"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/service/guardduty/waiter"
+)
+
+func resourceAwsGuardDutyPublishingDestination() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsGuardDutyPublishingDestinationCreate,
+		Read:   resourceAwsGuardDutyPublishingDestinationRead,
+		Update: resourceAwsGuardDutyPublishingDestinationUpdate,
+		Delete: resourceAwsGuardDutyPublishingDestinationDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"detector_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"destination_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  guardduty.DestinationTypeS3,
+				ValidateFunc: validation.StringInSlice([]string{
+					guardduty.DestinationTypeS3,
+				}, false),
```

Comment:
Nit: If this happens to be rebased with the main branch, we can take advantage of the new AWS Go SDK enumeration functionality here:

```suggestion
				ValidateFunc: validation.StringInSlice(guardduty.DestinationType_Values(), false),
```

Not a big deal though as we haven't begin the refactoring in #14601 yet.

Reply:
I haven't rebased recently, given the effort required to do so I'd prefer to get this merged first.